src/runtime/pkg/sev/sev.go

```diff
@@ -66,13 +66,27 @@ func fileSha256(filename string) (res [sha256.Size]byte, err error) {
 	return res, nil
 }
 
+// Calculate the SHA256 of the initrd file.  If an initrd file is not given,
+// return the SHA256 of the empty string.
```

Comment:
Can measured direct boot ever be used without an initrd, even outside of Kata? Does OVMF just skip checking components that don't have an entry in the table?

Reply:
Currently, QEMU always adds a hashes table with three entries (hashes): kernel, initrd, cmdline. If theres no `-initrd` argument for QEMU, it'll add SHA256("") for the initrd. I need to re-check the OVMF logic in this case.

Reply:
Ok I guess this just needs to match the QEMU implementation really. It's kind of a weird edgecase but I don't think it's dangerous.

Reply:
I re-checked [OVMF's behaviour](https://github.com/tianocore/edk2/blob/c5cf7f69c98baed40754ca4a821cb504fd5423cd/OvmfPkg/QemuKernelLoaderFsDxe/QemuKernelLoaderFsDxe.c#L1058-L1078): it'll read all three fw_cfg entries from QEMU (kernel, initrd, cmdline); all of them must exist and all of them must match the hash in the hashes table. QEMU is expected to fill SHA256("") for initrd (if not used) and SHA256("\0") for cmdline (if not used).